atlasaction/action.go

```diff
@@ -908,6 +905,113 @@ func (a *Actions) SchemaApply(ctx context.Context) error {
 	return nil
 }
 
+// schemaApprovedPlan returns the URL of the approved schema plan.
+func (a *Actions) schemaApplyWithApproval(ctx context.Context) ([]*atlasexec.SchemaApply, error) {
+	// Extract repo name from the URL it's provided. e.g. "atlas://repo-name?tag=tag-name"
+	var repo string
+	u, err := url.Parse(a.GetInput("to"))
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse the URL input: %w", err)
+	}
+	// If the repo can't be detected, we can't use the approval process.
+	if u.Scheme != "atlas" {
+		return nil, errors.New("approval process is not available for local schema")
+	}
```

Comment:
`schema plan` should work even if `--to` is not `atlas://`, but the `schema.repo` was set.